src/components/course/data/hooks.jsx

```diff
@@ -455,7 +455,7 @@ export const useTrackSearchConversionClickHandler = ({ href = undefined, eventNa
  * @returns Click handler function for clicks on enrollment buttons.
  */
 export const useOptimizelyEnrollmentClickHandler = ({ href, courseRunKey, userEnrollments }) => {
-  const hasNoExistingEnrollments = userEnrollments.length === 0;
+  const hasNoExistingEnrollments = userEnrollments?.length === 0;
```

Comment:
note: there is likely a conflict on this line due to a recent PR merging that also included a similar change